wordpress-mobile/AztecEditor-iOS
# Feature/452 add support for code element

Fixes #452 

This PR adds support for the [`<code>`](https://www.w3.org/TR/html5/textlevel-semantics.html#the-code-element) element.

While the majority of the code in this PR is pretty standard: adding a new character formatter with styling for the attribute (monospace font, background color), parsing and serialisation; There is something different on this attribute, the style it applies is not unique to this element.

So to implement we are using a custom attribute, .codeHTMLRepresentation that uses an HTMLRepresentation.

In order to make it work while editing, it need an addition to ```insertText``` method on the TextView to support the copy of custom attributes while writing text inside a code element.

At the moment the code is only making sure that the new .codeHTMLRepresentation is being copied but this could be expanded for new custom attributes.

To test:
 - Start the demo app
 - Open the demo with content
 - Check if the code attributes are being parsed ok
 - Write some text inside the code element
 - Check if style is still correct
 - Switch to HTML mode and check that the HTML is correctly generated.
